### PR TITLE
feat(kzip info): add proto output format to `kzip info`

### PR DESCRIPTION
**Pull request by justbuchanan**
proto makes it easier to work with this information internally and
having a defined schema is an improvement over hardcoding json key
strings in multiple places.

Example output:

```
$ kzip info --input test-infra.kzip --write_format proto

corpora: <
  key: "github.com/knative/test-infra"
  value: <
    files: <
      key: ""
      value: 143
    >
    files: <
      key: "go"
      value: 140
    >
    units: <
      key: "go"
      value: 49
    >
  >
>
corpora: <
  key: "golang.org"
  value: <
    files: <
      key: "go"
      value: 267
    >
  >
>
total_files: 550
total_units: 49
```

```
$ kzip info --input test-infra.kzip --write_format json | jq .
{
  "corpora": {
    "github.com/knative/test-infra": {
      "files": {
        "": 143,
        "go": 140
      },
      "units": {
        "go": 49
      }
    },
    "golang.org": {
      "files": {
        "go": 267
      }
    }
  },
  "total_files": 550,
  "total_units": 49
}

```

### CODE REVIEW

**Comment by justbuchanan** [comment]
Is there a better .proto file to put this new message in?

File: kythe/proto/storage.proto [hunk 1]
```diff
@@ -178,3 +178,21 @@ message ShardRequest {
   int64 index = 1;
   int64 shards = 2;
 }
+
+// KzipInfo contains a summary of the contents of a kzip. It provides a
+// breakdown of files and units by corpus and lanugage.
+message KzipInfo {
```
Comment (schroederc) [inline kythe/proto/storage.proto]: Probably better next to the compilation unit messages in `analysis.proto`

Comment (justbuchanan) [reply kythe/proto/storage.proto]: Done.

File: kythe/go/platform/tools/kzip/infocmd/infocmd.go [hunk 2]
```diff
@@ -21,20 +21,23 @@ import (
...
 	"encoding/json"
 	"flag"
-	"fmt"
+	"os"
+	"strings"
 
 	"kythe.io/kythe/go/platform/kzip"
 	"kythe.io/kythe/go/platform/vfs"
 	"kythe.io/kythe/go/util/cmdutil"
+	apb "kythe.io/kythe/proto/analysis_go_proto"
```
Comment (schroederc) [inline kythe/go/platform/tools/kzip/infocmd/infocmd.go]: Separate import group

File: kythe/go/platform/tools/kzip/infocmd/infocmd.go [hunk 3]
```diff
@@ -61,43 +65,44 @@ func (c *infoCommand) Execute(ctx context.Context, fs *flag.FlagSet, _ ...interf
...
+	if c.writeFormat != "json" && c.writeFormat != "proto" {
+		return c.Fail("Invalid --write_format. Can be 'json' or 'proto'.")
+	}
+
 	// Get file and unit counts broken down by corpus, language.
-	fileBreakdown := make(map[string]map[string]int)
-	unitBreakdown := make(map[string]map[string]int)
-	var totalFiles, totalUnits int
-	corpora := stringset.New()
+	kzipInfo := &apb.KzipInfo{Corpora: make(map[string]*apb.KzipInfo_CorpusInfo)}
```
Comment (schroederc) [inline kythe/go/platform/tools/kzip/infocmd/infocmd.go]:
```
corpusInfo := func(corpus string) *apb.KzipInfo_CorpusInfo {
  i := kzipInfo.Corpora[corpus]
  if i == nil {
    i = &apb.KzipInfo_CorpusInfo{
      Files: make(map[string]int32),
      Units: make(map[string]int32),
    }
    kzipInfo.Corpora[corpus] = i
  }
  return i
}

err = kzip.Scan(f, func(rd *kzip.Reader, u *kzip.Unit) error {
  kzipInfo.TotalUnits++
  corpusInfo(u.Proto.GetVName().GetCorpus()).Units[u.Proto.GetVName().GetLanguage()]++
  for _, ri := range u.Proto.RequiredInput {
    corpusInfo(ri.GetVName().GetCorpus()).Files[ri.GetVName().GetLanguage()]++
  }
}
```

File: kythe/go/platform/tools/kzip/infocmd/infocmd.go [hunk 4]
```diff
@@ -61,43 +65,44 @@ func (c *infoCommand) Execute(ctx context.Context, fs *flag.FlagSet, _ ...interf
...
-	out["unit_counts"] = unitBreakdown
-	out["file_counts"] = fileBreakdown
-	out["total_files"] = totalFiles
-	out["total_units"] = totalUnits
-	data, err := json.MarshalIndent(&out, "", "  ")
-	if err != nil {
-		return c.Fail("error marshaling json output: %v", err)
+	switch c.writeFormat {
+	case "json":
+		enc := json.NewEncoder(os.Stdout)
```
Comment (schroederc) [inline kythe/go/platform/tools/kzip/infocmd/infocmd.go]: Use `jsonpb`

File: kythe/go/platform/tools/kzip/infocmd/infocmd.go (hunk 3 above)
Comment (justbuchanan) [reply kythe/go/platform/tools/kzip/infocmd/infocmd.go]: Thanks!

**Pull request merged**